fix: deprecate options

UziTech opened the pull request:
> # 🚨  BREAKING CHANGE 🚨 
> do not submit until we are ready to release v5.0.0

## Description

Deprecate the following options:

| Option | Replacement|
|---------|---------------|
| `highlight`, `langPrefix`, and `callback`| [marked-highlight](https://www.npmjs.com/package/marked-highlight)|
|`mangle`| [marked-mangle](https://www.npmjs.com/package/marked-mangle)|
|`baseUrl`| [marked-base-url](https://www.npmjs.com/package/marked-base-url)|
|`smartypants`| [marked-smartypants](https://www.npmjs.com/package/marked-smartypants)|
|`xhtml`| [marked-xhtml](https://www.npmjs.com/package/marked-xhtml)|
|`headerIds` and `headerPrefix`| [marked-gfm-heading-id](https://www.npmjs.com/package/marked-gfm-heading-id)|

This will be a breaking change so should be done in the next major version.

This only deprecates the options it does not remove them yet. Ideally we will deprecate these in v5.0.0 and remove them in a later major release.

## Contributor

- [x] Test(s) exist to ensure functionality and minimize regression (if no tests added, list tests covering this PR); or,
- [ ] no tests required for this PR.
- [x] If submitting new feature, it has been documented in the appropriate places.

## Committer

In most cases, this should be a different person than the contributor.

- [ ] CI is green (no forced merge required).
- [ ] Squash and Merge PR following [conventional commit guidelines](https://www.conventionalcommits.org/).

joshbruce reviewed (approved):
Is it possible to make syntax highlighting completely opt in?

An extension that could be added like middleware. Thinking PHP [League\Commonmark](https://commonmark.thephpleague.com/2.4/extensions/overview/) and [Spatie\CommonmarkHighlighter](https://github.com/spatie/commonmark-shiki-highlighter).

I recognize with this solution users can use any highlighter. I'm wondering if it's possible for users to be able to use any highlighter *and* for Marked to not have to explicitly consider it.

UziTech commented:
> Is it possible to make syntax highlighting completely opt in?

Yes I could create a `marked-highlighter` extension that does essentially what this PR does (adds a walkTokens function).

That might be better. Then we can deprecate the highlighter option.

calculuschild commented:
I agree. Better to break it out into an extension.

UziTech commented:
I created [`marked-highlight`](https://www.npmjs.com/package/marked-highlight) and changed this so it only deprecates highlight and callback.

I updated the docs to recommend using `marked-highlight` for highlighting code.

styfle commented on docs/USING_ADVANCED.md:
```diff
@@ -76,6 +76,7 @@ Marked can be extended using [custom extensions](/using_pro#extensions). This is
 |[Emoji](https://github.com/UziTech/marked-emoji)|[`marked-emoji`](https://www.npmjs.com/package/marked-emoji)|Add emoji support like on GitHub|
 |[Extended Tables](https://github.com/calculuschild/marked-extended-tables)|[`marked-extended-tables`](https://www.npmjs.com/package/marked-extended-tables)|Extends the standard Github-Flavored tables to support advanced features: Column Spanning, Row Spanning, Multi-row headers|
 |[GFM Heading ID](https://github.com/markedjs/marked-gfm-heading-id)|[`marked-gfm-heading-id`](https://www.npmjs.com/package/marked-gfm-heading-id)|Use [`github-slugger`](https://github.com/Flet/github-slugger) to create the heading IDs and allow a custom prefix.|
+|[Highlight](https://github.com/UziTech/marked-highlight)|[`marked-highlight`](https://www.npmjs.com/package/marked-highlight)|Highlight code blocks|
```
Should we move this repo from https://github.com/UziTech to the org https://github.com/markedjs to make it look official?

UziTech replied:
I moved it to https://github.com/markedjs/marked-highlight

styfle commented on docs/USING_ADVANCED.md:
````diff
@@ -92,25 +93,9 @@ const inlineHtml = marked.parseInline('**strong** _em_');
 ```
 
-<h2 id="highlight">Asynchronous highlighting</h2>
+<h2 id="highlight">highlighting</h2>
````
```suggestion
<h2 id="highlight">Highlighting</h2>
```

UziTech commented:
I also deprecated the `mangle` option in this PR. I figure we should do all of the easy ones at once so users don't have to update their options too many times.

UziTech commented:
`headerIds` and `headerPrefix` can also be deprecated by the [marked-gfm-heading-id](https://github.com/markedjs/marked-gfm-heading-id) extension.

And I will create an extension for `baseUrl`.

UziTech commented:
I think I am done working on this. I all of the unnecessary options deprecated and moved to extensions. @joshbruce @calculuschild @styfle can you review this again?

styfle commented on src/helpers.js:
```diff
@@ -232,10 +232,42 @@ export function findClosingBracket(str, b) {
+  }
+
+  if (opt.headerIds || opt.headerPrefix) {
+    console.warn('marked(): headerIds and headerPrefix parameters are deprecated since version 5.0.0, should not be used and will be removed in the future. Use marked-gfm-heading-id from https://www.npmjs.com/package/marked-gfm-heading-id to add GFM ids to headers.');
```
```suggestion
    console.warn('marked(): headerIds and headerPrefix parameters are deprecated since version 5.0.0, should not be used and will be removed in the future. Instead use https://www.npmjs.com/package/marked-gfm-heading-id');
```

Can we change the format of all of these message to be a little shorter like this?
